apps/dashboard/tests/manage-workflows.e2e.ts

```diff
@@ -0,0 +1,172 @@
+
+  // add a step
+  await workflowEditorPage.addStepAsLast(StepTypeEnum.IN_APP);
+  let lastStep = await workflowEditorPage.getLastStep(StepTypeEnum.IN_APP);
+  await expect(lastStep).toBeVisible();
+  await workflowEditorPage.clickLastStep(StepTypeEnum.IN_APP);
+
+  // check the step config sidebar
+  const stepConfigSidebar = new StepConfigSidebar(page);
+  await expect(page).toHaveTitle(`Configure ${inAppStepName} | Novu Cloud Dashboard`);
+
+  // update the step name
+  const inAppStepNameUpdated = `${inAppStepName}-updated`;
+  await stepConfigSidebar.updateStepName({ oldStepName: inAppStepName, newStepName: inAppStepNameUpdated });
+  await stepConfigSidebar.configureTemplateClick();
+
+  const inAppStepEditor = new InAppStepEditor(page);
+  // Wait for navigation and check title
+  await page.waitForResponse('**/v2/workflows/**');
+  const title = await page.title();
+  await expect(title).toBe(`Edit ${inAppStepNameUpdated} | Novu Cloud Dashboard`);
+
+  // check the validation errors
+  await expect(await inAppStepEditor.getBodyValidationError()).toBeVisible();
+
+  await inAppStepEditor.fillForm({
+    subject,
+    body,
+    action: 'both',
+  });
+  // check the saved indicator
+  await expect(await inAppStepEditor.getSavedIndicator()).toBeVisible();
+
+  // check the preview
+  await inAppStepEditor.previewTabClick();
+  // TODO: add assertions for the primary and secondary actions
+  const previewElements = await inAppStepEditor.getPreviewElements();
+  await expect(await previewElements.subject).toContainText(subject);
+  await expect(await previewElements.body).toContainText(body);
+  await inAppStepEditor.close();
+
+  // add a second step
+  await workflowEditorPage.addStepAsLast(StepTypeEnum.IN_APP);
+
+  // check the step count
+  let stepCount = await workflowEditorPage.getStepCount(StepTypeEnum.IN_APP);
+  await expect(stepCount).toEqual(2);
+
+  // check the last step
+  lastStep = await workflowEditorPage.getLastStep(StepTypeEnum.IN_APP);
+  await expect(lastStep).toBeVisible();
+
+  // check the step config sidebar
+  await expect(page).toHaveTitle(`Edit ${inAppStepName} | Novu Cloud Dashboard`);
+  await inAppStepEditor.close();
+
+  // delete the step
+  await stepConfigSidebar.delete();
+
+  // check the step count
+  stepCount = await workflowEditorPage.getStepCount(StepTypeEnum.IN_APP);
+  await expect(stepCount).toEqual(1);
+
+  // go to the trigger tab
+  await workflowEditorPage.triggerTabClick();
+
+  // check the trigger workflow page
+  const triggerWorkflowPage = new TriggerWorkflowPage(page);
+  await expect(page).toHaveTitle(`Trigger ${workflowNameUpdated} | Novu Cloud Dashboard`);
+
+  // trigger the workflow
+  await triggerWorkflowPage.triggerWorkflowBtnClick();
+  const activityPanel = await triggerWorkflowPage.getActivityPanel();
+  await expect(activityPanel).toBeVisible();
+  await expect(activityPanel.locator('span').filter({ hasText: workflowNameUpdated })).toBeVisible();
```

Comment:
I suggest we run more assertions on the trigger page to ensure that the status is updated while running.

Reply:
I'll do that in the PR that I've opened for the activity panel fix: https://github.com/novuhq/novu/pull/7768